packages/uniforms-mui/src/SelectField.tsx

```diff
@@ -0,0 +1,225 @@
+          value={value ?? ''}
+        >
+          {allowedValues!.map(item => (
+            <FormControlLabel
+              control={
+                <Radio id={`${id}-${escape(item)}`} {...filteredProps} />
+              }
+              disabled={props.disableItem?.(item) || disabled}
+              key={item}
+              label={transform ? transform(item) : item}
+              value={item}
+            />
+          ))}
+        </RadioGroup>
+      ) : (
+        <FormGroup id={id}>
+          {allowedValues!.map(item => (
+            <FormControlLabel
+              control={
+                <SelectionControl
+                  checked={value.includes(item)}
+                  id={`${id}-${escape(item)}`}
+                  name={name}
+                  onChange={() =>
+                    disabled || readOnly || onChange(xor([item], value))
+                  }
+                  ref={inputRef}
+                  value={name}
+                  {...filteredProps}
+                />
+              }
+              disabled={props.disableItem?.(item) || disabled}
+              key={item}
+              label={transform ? transform(item) : item}
+            />
+          ))}
+        </FormGroup>
+      );
+
+    return wrapField(
+      { ...props, component: 'fieldset' },
+      (legend || label) && (
+        <FormLabel component="legend">{legend || label}</FormLabel>
+      ),
+      children,
+    );
+  }
+  const {
+    allowedValues,
+    disabled,
+    error,
+    errorMessage,
+    fieldType,
+    fullWidth = true,
+    helperText,
+    id,
+    InputLabelProps,
+    inputProps,
+    label,
+    labelProps,
+    margin = 'dense',
+    name,
+    native,
+    onChange,
+    placeholder,
+    readOnly,
+    required,
+    showInlineError,
+    transform,
+    variant,
+    textFieldProps,
+  } = props;
+
+  const Item = native ? 'option' : MenuItem;
+  const hasPlaceholder = !!placeholder;
+  const hasValue = value !== '' && value !== undefined;
+  const filteredProps = omit(filterDOMProps(props), [
+    'checkboxes',
+    'disableItem',
+    'fullWidth',
+    'helperText',
+    'margin',
+    'textFieldProps',
+    'variant',
+  ]);
+
+  return (
+    <TextField
+      disabled={disabled}
+      error={!!error}
+      fullWidth={fullWidth}
+      helperText={(error && showInlineError && errorMessage) || helperText}
+      InputLabelProps={{
+        shrink: !!label && (hasPlaceholder || hasValue),
+        ...labelProps,
+        ...InputLabelProps,
+      }}
+      label={label}
+      margin={margin}
+      onChange={event =>
+        disabled ||
+        readOnly ||
+        onChange(event.target.value !== '' ? event.target.value : undefined)
+      }
+      required={required}
+      select
+      SelectProps={{
+        displayEmpty: hasPlaceholder,
+        inputProps: { name, id, ...inputProps },
+        multiple: fieldType === Array || undefined,
+        native,
+        ...filteredProps,
```

Comment:
This one should be revisited, maybe for other fields as well
It seem to pass extra props to select dom element somehow
Warning:
```
Warning: React does not recognize the `InputLabelProps` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `inputlabelprops` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
div
```
Field with props:
```                
<AutoField
  name="name"
  size="small"
  sx={{ minWidth: 120 }}
  InputLabelProps={{ shrink: true }}
/>
```
Schema:
```
{
  type: 'string',
  enum: ['A', 'B', 'C', 'D'],
  title: 'Name',
}
```

Reply:
Might be good idea to investigate it here and reuse outcome approach for other themes

Reply:
@wight554 Can you file this issue? A brief description with an example or two will be more than enough.